src/java/com/google/cloud/bigquery/dwhassessment/hooks/logger/DatePartitionedRecordsWriterFactory.java

```diff
@@ -137,8 +152,16 @@ private String getDirForDate(LocalDate date) {
     return ISO_LOCAL_DATE.format(date);
   }
 
+  /**
+   * Next rollover time is at next configured interval or at the beginning of the next day,
+   * depending on what will happen earlier.
+   */
   private Instant calculateNextRolloverTime() {
-    return clock.instant().plus(1, ChronoUnit.DAYS).truncatedTo(ChronoUnit.DAYS);
+    Instant currentInstant = clock.instant();
+    Instant nextRollover = currentInstant.plus(rolloverInterval).truncatedTo(ChronoUnit.MINUTES);
+    Instant nextDay = currentInstant.plus(1, ChronoUnit.DAYS).truncatedTo(ChronoUnit.DAYS);
+
+    return nextDay.isAfter(nextRollover) ? nextRollover : nextDay;
```

Comment:
Optional: I would suggest to call the `min` method.

https://commons.apache.org/proper/commons-lang/javadocs/api-2.6/org/apache/commons/lang/ObjectUtils.html#min(java.lang.Comparable,%20java.lang.Comparable)

Reply:
Done